test/exchangeIssuance/exchangeIssuance-staged.spec.ts

```diff
@@ -0,0 +1,906 @@
+      }
+
+      it("should update the approvals correctly", async () => {
+        const token1 = subjectTokensToApprove[0];
+        const initUniswapToken1Allownace = await token1.allowance(exchangeIssuance.address, subjectUniswapRouter.address);
+        const initSushiswapToken1Allownace = await token1.allowance(exchangeIssuance.address, subjectSushiswapRouter.address);
+        const initIssuanceModuleToken1Allownace = await token1.allowance(exchangeIssuance.address, subjectBasicIssuanceModuleAddress);
+
+        const token2 = subjectTokensToApprove[1];
+        const initUniswapToken2Allownace = await token2.allowance(exchangeIssuance.address, subjectUniswapRouter.address);
+        const initSushiswapToken2Allownace = await token2.allowance(exchangeIssuance.address, subjectSushiswapRouter.address);
+        const initIssuanceModuleToken2Allownace = await token2.allowance(exchangeIssuance.address, subjectBasicIssuanceModuleAddress);
+
+        await subject();
+
+        const finalUniswapToken1Allownace = await token1.allowance(exchangeIssuance.address, subjectUniswapRouter.address);
+        const finalSushiswapToken1Allownace = await token1.allowance(exchangeIssuance.address, subjectSushiswapRouter.address);
+        const finalIssuanceModuleToken1Allownace = await token1.allowance(exchangeIssuance.address, subjectBasicIssuanceModuleAddress);
+
+        const finalUniswapToken2Allownace = await token2.allowance(exchangeIssuance.address, subjectUniswapRouter.address);
+        const finalSushiswapToken2Allownace = await token2.allowance(exchangeIssuance.address, subjectSushiswapRouter.address);
+        const finalIssuanceModuleToken2Allownace = await token2.allowance(exchangeIssuance.address, subjectBasicIssuanceModuleAddress);
+
+        expect(finalUniswapToken1Allownace.sub(initUniswapToken1Allownace)).eq(MAX_UINT_96);
+        expect(finalSushiswapToken1Allownace.sub(initSushiswapToken1Allownace)).eq(MAX_UINT_96);
+        expect(finalIssuanceModuleToken1Allownace.sub(initIssuanceModuleToken1Allownace)).eq(MAX_UINT_96);
+
+        expect(finalUniswapToken2Allownace.sub(initUniswapToken2Allownace)).eq(MAX_UINT_96);
+        expect(finalSushiswapToken2Allownace.sub(initSushiswapToken2Allownace)).eq(MAX_UINT_96);
+        expect(finalIssuanceModuleToken2Allownace.sub(initIssuanceModuleToken2Allownace)).eq(MAX_UINT_96);
+      });
+
+      context("when the set contains an external position", async () => {
+        beforeEach(async () => {
+          // Stick an external position into the Set
+        });
+
+        it("should revert", async () => {
+          // Verify contract reverts with the corerct error message
+          await expect(subject()).to.be.revertedWith("ExchangeIssuance: EXTERNAL_POSITIONS_NOT_ALLOWED");
+        });
+      });
+    });
+
+    describe("#approveSetToken", async () => {
+      let subjectSetToApprove: SetToken;
+      let subjectToken1: StandardTokenMock;
```

Comment:
These aren't parameters passed directly into the function, no need to preface as subject. Can also specify the token here (dai, wbtc)